plugin/grpctrace/interceptor.go

```diff
@@ -426,12 +426,11 @@ func peerInfoFromContext(ctx context.Context) []core.KeyValue {
 	return peerInfoFromTarget(p.Addr.String())
 }
 
-var fullMethodRegexp = regexp.MustCompile(`^/\S*\.(\S*)/\S*$`)
+var fullMethodRegexp = regexp.MustCompile(`^/(?:\S*/)?(?:\S*\.)?(\S*)/\S*$`)
 
 func serviceFromFullMethod(method string) string {
 	match := fullMethodRegexp.FindAllStringSubmatch(method, 1)
-
-	if len(match) != 1 && len(match[1]) != 2 {
+	if (len(match) == 0) || (len(match) > 1 && len(match[1]) != 2) {
```

Comment:
I think this could use an explanatory comment.  I'm not sure why `FindAllStringSubmatch()` is used and then this check added instead of using `FindStringSubmatch()` since it seems to only be concerned with the left-most submatch of the first capture group.

Reply:
I simplified the whole method a bit. It's (hopefully) more understandable now, as it uses FindStringSubmatch and then gets the sub-match found by the capture group, which is the second one in the list as there is one sub-expression only.